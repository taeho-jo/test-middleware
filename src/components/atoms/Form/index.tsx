import React from 'react';
// Styles
import { css } from '@emotion/react';

// Types
interface PropsType {
  width?: string;
  children: React.ReactNode;
  onSubmit: () => void;
}

const Form = ({ width = '100%', children, onSubmit }: PropsType) => {
  return (
    <form css={formStyle(width)} onSubmit={onSubmit}>
      {children}
    </form>
  );
};

export default Form;

const formStyle = width => css`
  width: ${width};
`;
