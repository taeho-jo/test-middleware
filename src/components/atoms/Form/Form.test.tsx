import React from 'react';
import { render } from '@testing-library/react';
import Form from './index';

describe('Form', () => {
  it('form', () => {
    const utils = render(<Form onSubmit={() => console.log('123')}>아아</Form>);
    expect(utils.container).toMatchSnapshot();
  });
});
