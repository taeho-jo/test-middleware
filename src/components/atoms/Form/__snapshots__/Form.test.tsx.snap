// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form form 1`] = `
<div>
  <form
    class="css-17ycjgs-formStyle-Form"
  >
    아아
  </form>
</div>
`;
